web/src/main/java/org/fao/geonet/services/metadata/MetadataCacheSeeder.java

```diff
@@ -0,0 +1,397 @@
+        Element md = Xml.loadString(getMetaData(dbms, curUuid), false);
+
+        root = root.addContent(md);
+
+        Element transformed = Xml.transform(root, xslPath);
+        FileUtils.writeStringToFile(new File(this.cachePath, curUuid + ".html"), Xml.getString(transformed));
+    }
+
+    private String getMetaData(Dbms dbms, String curUuid) throws Exception {
+        Element resp = dbms.select("SELECT data FROM metadata WHERE uuid = ? LIMIT 1; ", curUuid);
+        return resp.getChild("record").getChildText("data");
+    }
+
+    private Date getMetadataChangeDate(Dbms dbms, String curUuid) {
+        Date d = new Date();
+        try {
+            Element resp = dbms.select(
+                    "SELECT changedate FROM metadata WHERE uuid = ? LIMIT 1; ",
+                    curUuid);
+            String strDate = resp.getChild("record").getChildText("changedate");
+            String[] patterns = { DateFormatUtils.ISO_DATETIME_FORMAT
+                    .getPattern() };
+            strDate = strDate.substring(0, 19);
+            d = DateUtils.parseDate(strDate, patterns);
+        } catch (Exception e) {
+            // the main risk is to regenerate more often than planned the HTML doc.
+            Log.error(Geonet.CACHESEEDER,
+                    "Unable to get the Metadata change date (parse error), using current date.", e);
+        }
+        return d;
+    }
+
+    /**
+     * Removes a metadata which is still present in cache but no more public in the catalog.
+     * @param curUuid
+     */
+    private void cleanUpMetadata(String curUuid) {
+        try {
+            new File(this.cachePath, curUuid + ".html").delete();
+        } catch(Exception e) {
+            Log.error(Geonet.CACHESEEDER, "Unable to remove the Metadata", e);
+        }
+    }
+
+    /**
+     * Retrieves every public MD uuids.
+     * @return ArrayList an array list of strings.
+     * @throws SQLException 
+     */
+    private ArrayList<String> getPublicUuids(Dbms dbms) throws SQLException {
+        ArrayList<String> mdUuids = new ArrayList<String>();
+        // operationid = 0 (view)
+        // groupid = 1 (ALL)
+        // or harvested MDs (which are considered public)
+        Element resp = dbms.select("SELECT "
+                + "                        uuid "
+                + "                 FROM "
+                + "                        metadata"
+                + "                 FULL OUTER JOIN "
+                + "                        operationallowed "
+                + "                 ON "
+                + "                        metadata.id = operationallowed.metadataid "
+                + "                 WHERE "
+                + "                       (groupid = 1 AND operationid = 0 AND schemaid IN ('iso19139.pigma', 'iso19139', 'iso19139.fra')) "
+                + "                 OR "
+                + "                       (isharvested = 'y') ; " );
```

Comment:
you sure you want all harvested?  If you do then you can show metadata that users can't see if they do normal searches since harvested metadata is not necessarily public